### ISSUE TITLE
Type error: '?' expected.

### ISSUE THREAD

**Issue by seanlail** [prismicio/prismic-helpers#54]
### Versions

- @prismicio/helpers: 2.3.2
- node: v16.13.1
- typescript: 4.5.4

### Reproduction

I updated to 2.3.2 and when I run tsc I get this error:

```
./node_modules/@prismicio/helpers/dist/index.d.ts:399:2
Type error: '?' expected.

  397 |     embed_url: string;
  398 |     html: string | null;
> 399 | }) ? {
      |  ^
  400 |     [x: string]: never;
  401 | } | (Data & {
  402 |     embed_url: string;
```

### Steps to reproduce

- `npm i @prismicio/helpers@2.3.2`
- run tsc

### What is expected?

The build compiles

### What is actually happening?

Failed to compile

### Workaround

Use v2.3.1

**Comment by Clayakn** [comment]
Hey @seanlail  I was facing a similar issue when upgrading `@prismicio/client` from v5 to v6. I resolved it by upgrading the typescript library to the latest version which is `v4.7.4`. My build worked fine after that, hope this helps.

**Comment by lihbr** [comment]
Hey there, thank you for opening an issue! Thanks @Clayakn for providing support also!

Indeed between version 2.3.1 and 2.3.2 we bumped our TypeScript from version 4.6.3 to 4.7.4.

I ran some tests on my end and I'm surprised to see that the compiled library yields the same types though 🤔 So I'm a bit unsure what's the cause here (and I'd expect the TypeScript version not to yield such breaking changes also)

Does the solution provided by @Clayakn work for you?
If not, can you provide a minimal reproduction project so we can down further the root issue?

Let us know :)

**Comment by Clayakn** [comment]
@lihbr I don't understand the exact reasoning also, but I believe this is what is being addressed from typescript. 
https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#extends-constraints-on-infer-type-variables

**Comment by seanlail** [comment]
> Does the solution provided by @Clayakn work for you?

@lihbr Yes, works now. Thanks for the tip @Clayakn 🙇

**Comment by isaachinman** [comment]
Just got caught by this as well. @lihbr Just a heads up, the TypeScript project routinely introduces breaking changes on minor versions. Something to [watch out for](https://github.com/microsoft/TypeScript/wiki/Breaking-Changes).

**Comment by cptflammin** [comment]
Solved in the below case with latest helper
https://community.prismic.io/t/type-error-when-building-package-that-uses-prismicio-helpers/10938/10

**Comment by angeloashmore** [comment]
Hey everyone,

This issue should be fixed as of `v2.3.5`. For more details on the fix, see this PR: https://github.com/prismicio/prismic-helpers/pull/57

You can update to the latest version of `@prismicio/helpers` with the following command:

```sh
npm install @prismicio/helpers@latest
```

If this issue continues, please let us know and we can take another look. Thanks! 🙂
